src/data/nat/interval.lean

```diff
@@ -0,0 +1,85 @@
+/-
+Copyright (c) 2021 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.finset.interval
+
+/-!
+# Finite intervals of naturals
+
+This file proves that `ℕ` is a `locally_finite_order` and calculates the cardinality of its
+intervals as finsets and fintypes.
+
+## TODO
+
+Add all the `finset.Ico` stuff.
+-/
+
+open finset nat
+
+instance : locally_finite_order ℕ :=
+{ finset_Icc := λ a b, (list.range' a (b + 1 - a)).to_finset,
```

Comment:
I don't have a good reason but I generally regard `list` as a low-level CS thing that I try to avoid in the mathematics.

Is there a case for making the definitions like this instead:
```suggestion
{ finset_Icc := λ a b, (finset.range (b + 1)) \ (finset.range a),
```

Reply:
My reasoning is that I don't personally care about how `finset.Ixx` is defined, but if anyone is gonna do computation with that, they will want something that computes well. `list.range'` does exactly this. Further, all proofs involving peeking at the definition are one-liners.
If it's really using `list` that's bothering you, we could define `finset.range'`, but that's in my opinion pretty pointless as it will equal the current `finset.Ico`.

Reply:
You have persuaded me :-)